src/components/date-range/date-range.pw.tsx

```diff
@@ -0,0 +1,874 @@
+  });
+
+  test("should check the endDateProps prop", async ({ mount, page }) => {
+    await mount(
+      <DateRangeCustom
+        endDateProps={{
+          disabled: true,
+        }}
+      />
+    );
+
+    const endDateInput = getDataElementByValue(page, "input").nth(
+      END_DATE_RANGE_INDEX
+    );
+
+    await expect(endDateInput).toHaveAttribute("disabled", "");
+  });
+
+  (["top", "bottom", "left", "right"] as const).forEach((position) => {
+    test(`should check the tooltipPosition is set to ${position}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(
+        <DateRangeCustom
+          m={9}
+          tooltipPosition={position}
+          startError={testText}
+        />
+      );
+
+      const iconElement = icon(page);
+      await iconElement.nth(0).hover();
+      const getDataElementByValueElementTooltip = getDataElementByValue(
+        page,
+        "tooltip"
+      );
+      await expect(getDataElementByValueElementTooltip).toHaveAttribute(
+        "data-placement",
+        position
+      );
+
+      await page.hover("body"); // hover on body to close the tooltip
+    });
+  });
+});
+
+test.describe("Event tests for DateRange component", () => {
+  ([0, 1] as const).forEach((inputIndex) => {
+    test(`should call onChange callback when a clear event is triggered on ${inputIndex} DateRange input`, async ({
+      mount,
+      page,
+    }) => {
+      let callbackCount = 0;
+      await mount(
+        <DateRangeCustom
+          onChange={() => {
+            callbackCount += 1;
+          }}
+        />
+      );
+
+      const getDataElementByValueElementInputEqInputIndex = getDataElementByValue(
+        page,
+        "input"
+      ).nth(inputIndex);
+      await getDataElementByValueElementInputEqInputIndex.clear();
+
+      await expect(callbackCount).toEqual(1);
+    });
+  });
+
+  ([0, 1] as const).forEach((inputIndex) => {
+    test(`should call onChange callback when a type event is triggered on ${inputIndex} DateRange input`, async ({
+      mount,
+      page,
+    }) => {
+      let callbackCount = 0;
+      await mount(
+        <DateRangeCustom
+          onChange={() => {
+            callbackCount += 1;
+          }}
+        />
+      );
+
+      const getDataElementByValueElementInputEqInputIndex = getDataElementByValue(
+        page,
+        "input"
+      ).nth(inputIndex);
+      await getDataElementByValueElementInputEqInputIndex.type("01/10/2022");
+
+      await expect(callbackCount).toEqual(10);
+    });
+  });
+
+  test("should call onBlur callback when a blur event is triggered", async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <DateRangeCustom
+        onBlur={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+
+    const getDataElementByValueElementInputEq0 = getDataElementByValue(
+      page,
+      "input"
+    ).nth(0);
+    await getDataElementByValueElementInputEq0.press("Tab");
```

Comment:
**suggestion**: again could we just press tab on an input locator instead of declaring a const above